### PR TITLE
Re-former feature

### PR DESCRIPTION
**Pull request by moinkhanif**
Some of the features of this project include:

- User creation
- Entry validation
- User updation
- User update validation
- Error rendering

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class User < ApplicationRecord
+  validates :username, presence: true
+  validates :email, presence: true
+  validates :password, presence: true
```
Comment (suretrust) [inline app/models/user.rb:4]: You could add all on a straight line like => `validates :username, :email, :password, presence: true`

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+    if @user.update(user_params)
+
+    else
```
Comment (suretrust) [inline app/controllers/users_controller.rb:24]: `redirect_to edit_user_path` if the user was updated

File: app/views/users/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,49 @@
+<%
+=begin%>
+<h1>User Registration</h1>
+<form action='/users' method='POST'> <br>
```
Comment (suretrust) [inline app/views/users/new.html.erb:4]: This is missing `accept-charset="UTF-8"`. Should be `<form action='/users' method='POST' accept-charset="UTF-8">`

File: app/views/users/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  <label for='username'>Username</label> <br>
```
Comment (suretrust) [inline app/views/users/new.html.erb:5]: This should have the user object. More like => `<label for='user[username]'>Username</label>`

File: app/views/users/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  <input type='text' id='username' name='user[username]'> <br>
+  <label for='email'>Email</label> <br>
```
Comment (suretrust) [inline app/views/users/new.html.erb:7]: This should have the user object. More like => `<label for='user[email]'>Email</label>`

File: app/views/users/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  <input type='text' id='email' name='user[email]'> <br>
+  <label for='password'>Password</label> <br>
```
Comment (suretrust) [inline app/views/users/new.html.erb:9]: This should have the user object. More like => `<label for='user[password]'> Password </label>`

File: app/views/users/new.html.erb [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+  <input type='password' id='password' name='user[password]'> <br>
+  <input type="hidden" name="authenticity_token" value="<%= form_authenticity_token %>">
+  <input type='submit'>
+</form>
+<%
+=end%>
+
+<%
+=begin%>
+<%= form_tag("/users", method: "post") do%>
```
Comment (suretrust) [inline app/views/users/new.html.erb:19]: `method: "post"` is default value for this already so you don't need to define it again. It's only often needed for other actions like `DELETE`. It should be => `<%= form_tag("/users") do%>`

File: app/views/users/new.html.erb [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+<%= label_tag(:username, "Username") %>
+<%= text_field_tag(:username) %>
+<br>
+<%= label_tag(:email, "Email") %>
+<%= text_field_tag(:email) %>
```
Comment (suretrust) [inline app/views/users/new.html.erb:24]: This should be`<%= email_field_tag(:email) %>`.

File: app/views/users/new.html.erb [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+    <ul>
+    <% @user.errors.full_messages.each do |msg| %>
+      <li><%= msg %></li>
+    <% end %>
+    </ul>
+  </div>
+<% end %>
+<%= form_for @user do |f|%><br>
+Your username here: <%= f.text_field :username%><br>
+Your email here   :<%= f.text_field :email, placeholder: "email@example.com"%><br>
```
Comment (suretrust) [inline app/views/users/new.html.erb:46]: This should be `f.email_field`.

**Comment by moinkhanif** [comment]
Thank you for such a detailed review and also explaining each issue @suretrust ! 🎉

**Pull request merged**